src/Beman/Optional26/tests/optional_constexpr.t.cpp

```diff
@@ -0,0 +1,1261 @@
+            static_assert(b);
+        }
+
+        {
+            constexpr auto b = (o3 >= beman::optional26::nullopt);
+            static_assert(b);
+        }
+
+        {
+            constexpr auto b = (beman::optional26::nullopt >= o3);
+            static_assert(b);
+        }
+    }
+
+    //  SECTION("with T complex")
+    {
+        {
+            constexpr auto b = (!(o4 == Point{}));
+            static_assert(b);
+        }
+
+        {
+            constexpr auto b = (!(Point{} == o4));
+            static_assert(b);
+        }
+
+        {
+            constexpr auto b = (o4 != Point{});
+            static_assert(b);
+        }
+
+        {
+            constexpr auto b = (Point{} != o4);
+            static_assert(b);
+        }
+
+        {
+            constexpr auto b = (!(o4 < Point{}));
+            static_assert(b);
+        }
+
+        {
+            constexpr auto b = (Point{} < o4);
+            static_assert(b);
+        }
+
+        {
+            constexpr auto b = (o4 > Point{});
+            static_assert(b);
+        }
+
+        {
+            constexpr auto b = (!(Point{} > o4));
+            static_assert(b);
+        }
+
+        {
+            constexpr auto b = (!(o4 <= Point{}));
+            static_assert(b);
+        }
+
+        {
+            constexpr auto b = (Point{} <= o4);
+            static_assert(b);
+        }
+
+        {
+            constexpr auto b = (o4 >= Point{});
+            static_assert(b);
+        }
+
+        {
+            constexpr auto b = (!(Point{} >= o4));
+            static_assert(b);
+        }
+
+        {
+            constexpr auto b = (o4 == p4);
+            static_assert(b);
+        }
+
+        {
+            constexpr auto b = (p4 == o4);
+            static_assert(b);
+        }
+
+        {
+            constexpr auto b = (!(o4 != p4));
+            static_assert(b);
+        }
+
+        {
+            constexpr auto b = (!(p4 != o4));
+            static_assert(b);
+        }
+
+        {
+            constexpr auto b = (!(o4 < p4));
+            static_assert(b);
+        }
+
+        {
+            constexpr auto b = (!(p4 < o4));
+            static_assert(b);
+        }
+
+        {
+            constexpr auto b = (!(o4 > p4));
+            static_assert(b);
+        }
+
+        {
+            constexpr auto b = (!(p4 > o4));
+            static_assert(b);
+        }
+
+        {
+            constexpr auto b = (o4 <= p4);
+            static_assert(b);
+        }
+
+        {
+            constexpr auto b = (p4 <= o4);
+            static_assert(b);
+        }
+
+        {
+            constexpr auto b = (o4 >= p4);
+            static_assert(b);
+        }
+
+        {
+            constexpr auto b = (p4 >= o4);
+            static_assert(b);
+        }
+    }
+    return true;
+}
+
+constexpr bool checkTestComparison = testComparisons();
+static_assert(checkTestComparison);
+
+TEST(OptionalConstexprTest, RangeTest) {
```

Comment:
Should this test be added at the end of optional_range_support.t.cpp file?

Also, I was expecting do have a constexpr function which creates 2 ranges, iterates over them and compute something, and check the result at compile time. is this mixt test useful/complete?

Reply:
Ok. let's keep it as it is.